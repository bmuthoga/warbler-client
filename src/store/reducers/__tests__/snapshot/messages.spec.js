import messagesReducer from '../../messages'
import { LOAD_MESSAGES, REMOVE_MESSAGE } from '../../../actionTypes'

describe('Messages Reducer Snapshot Tests', () => {
  it('should return the initial state', () => {
    expect(messagesReducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle LOAD_MESSAGES', () => {
    const messages = [
      {
        _id: '87y42f49hg94h9g',
        text: 'First sample message',
        user: {
          _id: '938490u98r4jv94jvj8',
          username: 'firstTestUser',
          profileImageUrl: 'https://lorempixel.com/200/200/'
        },
        createdAt: '2019-11-23T16:45:46.174Z',
        updatedAt: '2019-11-23T16:45:46.174Z'
      },
      {
        _id: '9485g0984g04g0094g',
        text: 'Second sample message',
        user: {
          _id: '948v0845v704788r7v879r',
          username: 'secondTestUser',
          profileImageUrl: 'https://lorempixel.com/200/200/'
        },
        createdAt: '2019-11-23T16:41:16.340Z',
        updatedAt: '2019-11-23T16:41:16.340Z'
      }
    ]
    expect(messagesReducer(undefined, { type: LOAD_MESSAGES, messages })).toMatchSnapshot()
  })

  it('should handle REMOVE_MESSAGE', () => {
    const state = [
      {
        _id: '87y42f49hg94h9g',
        text: 'First sample message',
        user: {
          _id: '938490u98r4jv94jvj8',
          username: 'firstTestUser',
          profileImageUrl: 'https://lorempixel.com/200/200/'
        },
        createdAt: '2019-11-23T16:45:46.174Z',
        updatedAt: '2019-11-23T16:45:46.174Z'
      },
      {
        _id: '9485g0984g04g0094g',
        text: 'Second sample message',
        user: {
          _id: '948v0845v704788r7v879r',
          username: 'secondTestUser',
          profileImageUrl: 'https://lorempixel.com/200/200/'
        },
        createdAt: '2019-11-23T16:41:16.340Z',
        updatedAt: '2019-11-23T16:41:16.340Z'
      }
    ]
    expect(messagesReducer(state, { type: REMOVE_MESSAGE, id: state[1]._id })).toMatchSnapshot()
  })
})
