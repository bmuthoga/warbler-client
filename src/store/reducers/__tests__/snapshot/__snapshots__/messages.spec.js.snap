// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages Reducer Snapshot Tests should handle LOAD_MESSAGES 1`] = `
Array [
  Object {
    "_id": "87y42f49hg94h9g",
    "createdAt": "2019-11-23T16:45:46.174Z",
    "text": "First sample message",
    "updatedAt": "2019-11-23T16:45:46.174Z",
    "user": Object {
      "_id": "938490u98r4jv94jvj8",
      "profileImageUrl": "https://lorempixel.com/200/200/",
      "username": "firstTestUser",
    },
  },
  Object {
    "_id": "9485g0984g04g0094g",
    "createdAt": "2019-11-23T16:41:16.340Z",
    "text": "Second sample message",
    "updatedAt": "2019-11-23T16:41:16.340Z",
    "user": Object {
      "_id": "948v0845v704788r7v879r",
      "profileImageUrl": "https://lorempixel.com/200/200/",
      "username": "secondTestUser",
    },
  },
]
`;

exports[`Messages Reducer Snapshot Tests should handle REMOVE_MESSAGE 1`] = `
Array [
  Object {
    "_id": "87y42f49hg94h9g",
    "createdAt": "2019-11-23T16:45:46.174Z",
    "text": "First sample message",
    "updatedAt": "2019-11-23T16:45:46.174Z",
    "user": Object {
      "_id": "938490u98r4jv94jvj8",
      "profileImageUrl": "https://lorempixel.com/200/200/",
      "username": "firstTestUser",
    },
  },
]
`;

exports[`Messages Reducer Snapshot Tests should return the initial state 1`] = `Array []`;
